app/assets/javascripts/discourse/app/lib/do-not-disturb.js

```diff
@@ -0,0 +1,39 @@
+import DiscourseURL from "discourse/lib/url";
+import { Promise } from "rsvp";
+import { ajax } from "discourse/lib/ajax";
+
+export function publishDoNotDisturbOnFor(user, duration) {
+  return new Promise(function (resolve, reject) {
```

Comment:
You shouldn't need to return `Promise`s from these two functions, since `ajax` returns a promise already. The same code should work with `return ajax(...)`